src/Client.php

```diff
@@ -272,7 +273,126 @@ public function getAsset($assetId) {
       ['headers' => $this->getDefaultHeaders()]
     );
 
-    return (string) $response->getBody();
+    $asset = Asset::fromJson((string) $response->getBody());
+
+    return $asset;
+  }
+
+  /**
+   * Gets presigned url from AWS S3.
+   *
+   * @param array $file_data
+   *   The file data required by Webdam
+   *   (filesize, folder_id, filename, contenttype).
+   *
+   * @return mixed
+   *   Presigned url needed for next step + PID.
+   */
+  protected function getPresignUrl(array $file_data) {
+    $this->checkAuth();
+
+    $response = $this->client->request(
+      "GET",
+      $this->baseUrl . '/ws/awss3/generateupload',
+      [
+        'headers' => $this->getDefaultHeaders(),
+        'query' => $file_data,
+      ]
+    );
+    return json_decode($response->getBody());
+  }
+
+  /**
+   * Uploads file to Webdam AWS S3.
+   *
+   * @param mixed $presignedUrl
+   *   The presigned URL we got in previous step from AWS.
+   * @param string $file_uri
+   *   The file URI.
+   * @param string $file_type
+   *   The File Content Type.
+   * @return array
+   *   Response Status 100 / 200
+   */
+  protected function uploadPresigned($presignedUrl, $file_uri, $file_type) {
+    $this->checkAuth();
+
+    $response = $this->client->request(
+      "PUT",
+      $presignedUrl, [
+        'headers' => ['Content-Type' => $file_type],
+        'body' => $file_uri,
```

Comment:
mhh... not sure why?
In there we are sending the file to the presigned url we got from AWS, also sending the Conten-Type in header.
API docs https://www.damsuccess.com/hc/en-us/articles/202134055-REST-API#uploadasset 

Let me know if I'm missing something!